comp/otelcol/otlp/components/exporter/serializerexporter/factory.go

```diff
@@ -132,6 +141,13 @@ func newFactoryForAgentWithType(
 		store:        store,
 	}
 
+	if reporter != nil {
+		// reporter is initialized in datadogexporter.NewFactory in DDOT, no need to initialize it again
+		f.onceReporter.Do(func() {
+			f.reporter = reporter
+		})
```

Comment:
Does this mean that the only case where we would reach `f.reporter = reporter` is when we pass in a reporter in OSS collector?

Reply:
Not really, this is solely for the DD exporter in otel-agent. The upstream OSS DD exporter goes through a different code path: https://github.com/DataDog/datadog-agent/blob/main/comp/otelcol/otlp/components/exporter/serializerexporter/factory.go#L181-L196